### PR TITLE
Fixes issue where explicit non-pvc volume in template was ignored

### PR DESCRIPTION
**Pull request by pebrc**
This PR
* fixes an issue where any non-PVC volumes in the pod spec would be ignored and a default pvc sourced volume would be created all the time. 
* adjusts existing unit tests
* adds two e2e tests 
    * one for the PVC default mechanism
    * one testing an explicit override via the pod template with emptyDir

### CODE REVIEW

File: operators/test/e2e/volume_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	k := helpers.NewK8sClientOrFatal()
+
+	initStack := stack.NewStackBuilder("test-es-default-pvc").
+		WithESMasterNodes(1, stack.DefaultResources)
+
+	helpers.TestStepList{}.
+		WithSteps(stack.InitTestSteps(initStack, k)...).
+		WithSteps(stack.CreationTestSteps(initStack, k)...).
+		WithSteps(stack.ESClusterChecks(initStack.Elasticsearch, k)...).
+		WithSteps(stack.CheckDefaultPVC(initStack.Elasticsearch, k)).
```
Comment (sebgl) [inline operators/test/e2e/volume_test.go]: should we include this call in `CheckStackSteps()`, to be performed for every test suite (default case), and add a single new test for empty dirs instead of adding 2 new tests?

Comment (pebrc) [reply operators/test/e2e/volume_test.go]: Sgtm, will add another commit

Comment (pebrc) [reply operators/test/e2e/volume_test.go]: @sebgl if you could re-LGTM and check if I understood your idea correctly ....

File: operators/test/e2e/volume_test.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	k := helpers.NewK8sClientOrFatal()
+
+	initStack := stack.NewStackBuilder("test-es-explicit-empty-dir").
+		WithESMasterNodes(1, stack.DefaultResources).
+		WithEmptyDirVolumes()
+
+	helpers.TestStepList{}.
+		WithSteps(stack.InitTestSteps(initStack, k)...).
+		WithSteps(stack.CreationTestSteps(initStack, k)...).
+		WithSteps(stack.ESClusterChecks(initStack.Elasticsearch, k)...).
```
Comment (sebgl) [inline operators/test/e2e/volume_test.go]: I think `ESClusterChecks` is already called from `CreationTestSteps`?
Also since `CreationTestSteps ` calls `K8sStackChecks`, will tests look for a PV in the empty dir scenario?

I was thinking we would have a single `checkVolumes` step that, depending on the provided stack spec, would either look for EmptyDir or PVCs.

Comment (pebrc) [reply operators/test/e2e/volume_test.go]: Oh, yikes, I did not realise how deeply nested the russian dolls were...

Comment (pebrc) [reply operators/test/e2e/volume_test.go]: @sebgl I gave it another try
